uxarray/grid.py

```diff
@@ -372,6 +373,82 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
 
         return integral
 
+    # A one-time-used helper function that will only be used in the build_face_edges_connectivity map() function usage
+    def __replace_fill_val(self, row, index, rep_val):
+        row[index] = rep_val
+        return row
+
+    def build_face_edges_connectivity(self):
+        """A DataArray of indices indicating edges that are neighboring each
+        face.
+
+        Notes
+        -----
+        This function will add `Grid.ds.Mesh2_face_edges` to the `Grid` class, which is an integer
+        DataArray of size (nMesh2_face, MaxNumNodesPerFace)
+        """
+        mesh2_face_nodes = self.Mesh2_face_nodes.values
+        n = self.nMesh2_face
+        m = self.nMaxMesh2_face_nodes
+
+        # First identify the _FillValue in the mesh2_face_nodes
+        # We will unify the _FillValue used in this function as -1:
+        if np.isnan(_FillValue):
+            mesh2_face_nodes = np.nan_to_num(mesh2_face_nodes, nan=-1)
+        elif _FillValue is not -1:
+            mesh2_face_nodes[mesh2_face_nodes == _FillValue] = -1
+
+        # Then do the padding for each face to close the polygon
+        closed = np.full((n, m + 1), -1, dtype=np.intp)
+        closed[:, :-1] = np.array(mesh2_face_nodes, dtype=np.intp)
+        # We only want the index of first occurrence of -1
+        first_fill_value_index = np.argmax(closed == -1, axis=1)
+        first_node = mesh2_face_nodes[:, 0]
+
+        # Now replace the first -1 at each row to be the first node
+        pad_results = np.array(
+            list(
+                map(self.__replace_fill_val, closed, first_fill_value_index,
+                    first_node)))
+
```

Comment:
Can eliminate the need for the `__replace_fill_val` helper function and mapping by indexing `closed` as a 1D array to replace values (doing this operating on 2D arrays in numpy is messy and I haven't been able to get it to work well). I've tested the following approach and it is equivalent to what you have.

```Python
# Create 1D index into a 2D array
first_fill_idx = (m+1)*np.arange(0, n) + first_fill_value_index

# replace all values at 1D index with the first_node 
np.put(closed.ravel(), first_fill_idx, first_node)

pad_results = closed

Reply:
Changed, thank you very much